tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ServiceStackRedisTests.cs

```diff
@@ -26,68 +30,52 @@ public ServiceStackRedisTests(ITestOutputHelper output)
-                    Assert.Equal(port, DictionaryExtensions.GetValueOrDefault(span.Tags, "out.port"));
-                    Assert.False(span.Tags?.ContainsKey(Tags.Version), "External service span should not have service version tag.");
-                }
-
-                var expectedFromOneRun = new TupleList<string, string>
-                {
-                    { "SET", $"SET {TestPrefix}ServiceStack.Redis.INCR 0" },
-                    { "PING", "PING" },
-                    { "DDCUSTOM", "DDCUSTOM COMMAND" },
-                    { "ECHO", "ECHO Hello World" },
-                    { "SLOWLOG", "SLOWLOG GET 5" },
-                    { "INCR", $"INCR {TestPrefix}ServiceStack.Redis.INCR" },
-                    { "INCRBYFLOAT", $"INCRBYFLOAT {TestPrefix}ServiceStack.Redis.INCR 1.25" },
-                    { "TIME", "TIME" },
-                    { "SELECT", "SELECT 0" },
-                };
-
-                var expected = new TupleList<string, string>();
-                expected.AddRange(expectedFromOneRun);
-                expected.AddRange(expectedFromOneRun);
-#if NETCOREAPP3_1_OR_GREATER
-                expected.AddRange(expectedFromOneRun); // On .NET Core 3.1 and .NET 5 we run the routine a third time
-#endif
+                var settings = VerifyHelper.GetSpanVerifierSettings();
+                settings.UseFileName($"{nameof(ServiceStackRedisTests)}.RunServiceStack");
+                settings.DisableRequireUniquePrefix();
+                settings.AddSimpleScrubber($" {TestPrefix}ServiceStack.Redis.", " ServiceStack.Redis.");
+                settings.AddSimpleScrubber($"out.host: {host}", "out.host: servicestackredis");
+                settings.AddSimpleScrubber($"out.port: {port}", "out.port: 6379");
 
-                Assert.Equal(expected.Count, spans.Count);
-
-                for (int i = 0; i < expected.Count; i++)
+                // The test application runs the same RunServiceStack method X number of times
+                // Use the snapshot to verify each run of RunServiceStack, instead of testing all of the application spans
+                for (int i = 0; i < numberOfRuns; i++)
                 {
-                    var e1 = expected[i].Item1;
-                    var e2 = expected[i].Item2;
-
-                    var a1 = i < spans.Count
-                                 ? spans[i].Resource
-                                 : string.Empty;
-                    var a2 = i < spans.Count
-                                 ? DictionaryExtensions.GetValueOrDefault(spans[i].Tags, "redis.raw_command")
-                                 : string.Empty;
-
-                    Assert.True(e1 == a1, $@"invalid resource name for span #{i}, expected ""{e1}"", actual ""{a1}""");
-                    Assert.True(e2 == a2, $@"invalid raw command for span #{i}, expected ""{e2}"" != ""{a2}""");
+                    var routineSpans = spans.GetRange(i * expectedSpansPerRun, expectedSpansPerRun).AsReadOnly();
```

Comment:
I guess this assumes the spans are always received in the same order they're sent. AFAICT there aren't any async commands (really?!) so that seems like a reasonable assumption, but just wanted to confirm 🙂

Reply:
Yep, none of the methods are async so we're also asserting that the span order is always the same